server/services/v1/query_runner.go

```diff
@@ -736,6 +742,31 @@ func (runner *SearchQueryRunner) getFieldSelection(collFields []*schema.Queryabl
 	return factory, nil
 }
 
+func (runner *SearchQueryRunner) getSortOrdering(collFields []*schema.QueryableField) (*qsearch.Ordering, error) {
+	ordering, err := qsearch.UnmarshalSort(runner.req.GetSort())
+	if err != nil || ordering == nil {
+		return nil, err
+	}
+
+	for _, sf := range *ordering {
+		found := false
+		for _, cf := range collFields {
```

Comment:
not for this diff, but probably we can have a `GetField(name string)` method on the collection. Then the few checks that we are doing in the case of facet search, sorting, etc can be unified by just getting this field from the collection and if not present returning an error otherwise check if the field can be used to serve the request.